synapse/handlers/oidc.py

```diff
@@ -246,6 +253,78 @@ async def handle_oidc_callback(self, request: SynapseRequest) -> None:
 
         await oidc_provider.handle_oidc_callback(request, session_data, code)
 
+    async def handle_backchannel_logout(self, request: SynapseRequest) -> None:
+        """Handle an incoming request to /_synapse/client/oidc/backchannel_logout
+
+        This extracts the logout_token from the request and try to figure out
+        from whom it is coming from. This works by matching the iss claim with
+        the issuer and the aud claim with the client_id.
+
+        Since at this point we don't know who signed the JWT, we can't just
+        decode it automatically, we have to manually parse the JWT to extract
+        both claims.
+
+        Args:
+            request: the incoming request from the browser.
+        """
+        logout_token = parse_string(request, "logout_token")
+        if logout_token is None:
+            raise SynapseError(400, "Missing logout_token in request")
+
+        # A JWT looks like this:
+        #    header.payload.signature
+        # where all parts are encoded with urlsafe base64.
+        # The aud and iss claims we care about are in the payload part, which
+        # is a JSON object.
+        try:
+            # This raises if there are too many or not enough segments in the token
+            header, payload, signature = logout_token.rsplit(".", 4)
```

Comment:
Why the `maxsplit=4`? I only see 3 parts listed above, so I think I would expect a `maxsplit=2`?

Although it seems like we really expect the exact number of results to be 3, so I think we can just leave out the `maxsplit`?